### PR TITLE
Fix #74 host value in openshift_option should be used effectively

### LINKED ISSUE #74: host value in openshift_option should be used effectively
@hferentschik pointed out below situation and we should have better use-case for `host` value in openshift_option.

why do you hardcode xip.io?  What if someone wants to go the Landrush route?

Is the problem not in openshift_option. There we just use the hostname, but it should be something like 10.1.2.2.xip.io or $(hostname).10.1.2.2.xip.io. In the latter case the hub host would be something like hub.openshift.$(hostname).10.1.2.2.xip.io.

Discussion : https://github.com/projectatomic/adb-utils/pull/67

### PR DESCRIPTION
**Pull request by praveenkumar**
we should not  hardcode xip.io because what if someone wants to go the Landrush route?

Is the problem not in openshift_option. There we just use the hostname, but it should be something like 10.1.2.2.xip.io or $(hostname).10.1.2.2.xip.io. In the latter case the hub host would be something like hub.openshift.$(hostname).10.1.2.2.xip.io

### CODE REVIEW

**Comment by praveenkumar** [comment]
@hferentschik please review and provide your feedback.

**Comment by praveenkumar** [comment]
> If you go for OPENSHIFT_SUBDOMAIN, the question arises whether we still need a generic OPTIONS atm. It seems we don't.

Yes, I think that will work.

**Comment by hferentschik** [comment]
> Yes, I think that will work.

sweet

**Comment by praveenkumar** [comment]
@hferentschik  Done, review now.

File: services/openshift/openshift_option [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # /etc/sysconfig/openshift_options
 
 # Modify these options if you want to change openshift hostname
-OPTIONS="-host $(hostname)"
+OPENSHIFT_SUBDOMAIN="$(hostname).$(/opt/adb/openshift/get_ip_address).xip.io"
```
Comment (hferentschik) [inline services/openshift/openshift_option:4]: +1, exactly what I had in mind

File: services/openshift/scripts/openshift_provision [hunk 2]
```diff
@@ -36,8 +33,10 @@ wait_for_openshift_api
 # Note: oc and oadm are symlinks to openshift
 binaries=(openshift oc oadm)
 for n in ${binaries[@]}; do
-  echo "[INFO] Copying ${n} binary to VM"
-  docker cp openshift:/usr/bin/${n} /usr/bin/${n}
+  if [ ! -f /usr/bin/${n} ]; then
```
Comment (hferentschik) [inline services/openshift/scripts/openshift_provision:37]: +1

**Comment by hferentschik** [comment]
>  Done, review now.

Looking good :-)

**Comment by bexelbie** [comment]
LGTM

**Pull request merged**